src/dune_rules/lock_dir.mli

```diff
@@ -4,8 +4,10 @@ module Pkg = Dune_pkg.Lock_dir.Pkg
 type t := Dune_pkg.Lock_dir.t
 
 val get : Context_name.t -> t Memo.t
+val of_dev_tool : string -> t Memo.t
```

Comment:
It's not clear what the meaning of the `string` argument to this function is. This needs a comment or better yet using a more meaningful type than `string` for the argument. (Same for `dev_tool_pat` below)